google/cloud/bigquery/table.py

```diff
@@ -1810,7 +1810,7 @@ def to_arrow(
             if owns_bqstorage_client:
                 bqstorage_client._transport.grpc_channel.close()
 
-        if record_batches:
+        if record_batches and bqstorage_client is not None:
```

Comment:
Curious what bug did this extra check for the BQ Storage client fix?

Reply:
When `bqstorage_client` is `None`, we're using the REST API, which doesn't add arrow extension metadata.

I improved the comment.